### PR TITLE
ref(integrations): Azure DevOps add logging to Webhook.

### PR DESCRIPTION
**Pull request by lauryndbrown**

### CODE REVIEW

File: src/sentry/integrations/vsts/webhooks.py [hunk 1]
```diff
@@ -28,29 +28,78 @@ def dispatch(self, request, *args, **kwargs):
...
-                provider='vsts',
-                external_id=data['resourceContainers']['collection']['id'],
+        try:
+            event_type = data['eventType']
+            external_id = data['resourceContainers']['collection']['id']
+        except KeyError as e:
+            logger.info(
+                'vsts.invalid-webhook-payload',
+                extra={
+                    'error': e.message,
```
[the remark this answers is not in the document]
Comment (lauryndbrown) [reply src/sentry/integrations/vsts/webhooks.py]: Yikes.... ok

File: src/sentry/integrations/vsts/webhooks.py [hunk 2]
```diff
@@ -120,11 +125,24 @@ def handle_assign_to(self, integration, external_issue_key, assigned_to):
...
+                        'assigned_to_values': assigned_to,
+                        'external_issue_key': external_issue_key,
+                    }
+                )
+                return  # TODO(lb): return if cannot parse email?
             assign = True
         else:
             email = None
             assign = False
+        # TODO(lb): should there be a logging statement here? or a try-catch?
```
Comment (markstory) [inline src/sentry/integrations/vsts/webhooks.py]: Does this work interact with external systems? If no then exceptions could be valid problems with sentry and not something we would want to catch and log.

File: src/sentry/integrations/vsts/webhooks.py [hunk 3]
```diff
@@ -28,32 +30,84 @@ def dispatch(self, request, *args, **kwargs):
...
-                raven.captureException()
+                logger.info(
+                    'vsts.invalid-webhook-secret',
+                    extra={
+                        'event_type': event_type,
+                        'integration_id': integration.id
+                    }
+                )
+                # TODO(lb): I'm not sure we should even return anything negative to vsts
+                # lest it disable the webhook
```
[the remark this answers is not in the document]
Comment (lauryndbrown) [reply src/sentry/integrations/vsts/webhooks.py]: No it wouldn't. Sounds good then.

File: src/sentry/integrations/vsts/webhooks.py [hunk 4]
```diff
@@ -28,32 +30,84 @@ def dispatch(self, request, *args, **kwargs):
...
+        except KeyError as e:
+            logger.info(
+                'vsts.missing-webhook-secret',
+                extra={
+                    'error': six.text_type(e),
+                    'integration_id': integration.id,
+                }
+            )
+
+        assert integration_secret == webhook_payload_secret
```
[the remark this answers is not in the document]
Comment (lauryndbrown) [reply src/sentry/integrations/vsts/webhooks.py]: :O ok I'll add that.

File: src/sentry/integrations/vsts/webhooks.py [hunk 5]
```diff
@@ -99,9 +166,11 @@ def handle_status_change(self, integration, external_issue_key,
                 'old_state': status_change['oldValue'],
                 'project': project,
             }
+
             installation.sync_status_inbound(external_issue_key, data)
 
     def parse_email(self, email):
+        # TODO(lb): hmm... this looks brittle to me
         return EMAIL_PARSER.search(email).group(1)
```
[the remark this answers is not in the document]
Comment (lauryndbrown) [reply src/sentry/integrations/vsts/webhooks.py]: Well I added the logging and error handling above. Is that ok? Checkout line 128 to 140.

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**